lib/java/desktop-environment/src/main/java/org/enso/desktopenvironment/JnaTrash.java

```diff
@@ -0,0 +1,34 @@
+package org.enso.desktopenvironment;
+
+import com.sun.jna.platform.FileUtils;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * JNA trash implementation calling native platform-dependent APIs.
+ *
+ * <p>Native {@link FileUtils} are implemented for Windows and macOS platforms.
+ */
+final class JnaTrash implements Trash {
+
+  @Override
+  public boolean isSupported() {
+    return FileUtils.getInstance().hasTrash();
+  }
+
+  @Override
+  public boolean moveToTrash(Path path) {
+    if (Files.exists(path) && isSupported()) {
+      try {
+        FileUtils.getInstance().moveToTrash(path.toFile());
```

Comment:
The Windows implementation is defined at [W32FileUtils](https://github.com/java-native-access/jna/blob/master/contrib/platform/src/com/sun/jna/platform/win32/W32FileUtils.java#L31) and it seems to call into:
- [Shell API](https://github.com/java-native-access/jna/blob/master/contrib/platform/src/com/sun/jna/platform/win32/Shell32.java#L48)
- [SHFileOperation](https://github.com/java-native-access/jna/blob/master/contrib/platform/src/com/sun/jna/platform/win32/Shell32.java#L93)